pages/video-detail/video-detail.controller.js

```diff
@@ -0,0 +1,44 @@
+angular.module('app')
+	.controller('VideoDetailController', ['$stateParams', 'session', 'video', 'user', '$timeout', VideoDetail]);
+
+function VideoDetail($stateParams, session, video, user, $timeout) {
+	var self = this;
+
+	self.id = $stateParams.id;
+	self.code = '';
+	self.loading = true;
+	self.loadingAuthor = true;
+	self.session = session; 
+
+	self.video = video;
+	self.user = user;
+
+	self.showMore = showMore();
+
+	if(self.video && self.video.data && self.video.data.id !== self.id) {
+		self.video.reset();
+	}
+	
+	// ToDo: Check invalid video :S
```

Comment:
we should have a 404 directive, template or redirect to not get lazy with this error checks in the begining. If later on we want to treat error more specifically, we can override behaviour, but this way, we just use the default error handler.

Reply:
Definitely yes. But, when talking with @PabloFMM and Juan, It seemed better to have different 404 implementations for each type (ie, "Video not exist", "User not exist"...)

Reply:
I gues that could be also some kind of parameter of the template/directive, etc